Detectors/MUON/MCH/Simulation/src/Digitizer.cxx

```diff
@@ -88,15 +88,19 @@ void Digitizer::process(const std::vector<Hit> hits, std::vector<Digit>& digits)
     int labelIndex = mMCTruthContainer.getIndexedSize();
     //index for this hit
     int detID = hit.GetDetectorID();
-    processHit(hit, detID, mEventTime, labelIndex);
-    MCCompLabel label(hit.GetTrackID(), mEventID, mSrcID);
-    mMCTruthContainer.addElementRandomAccess(labelIndex, label);
+    int ndigits =  processHit(hit, detID, mEventTime);
+    //TODO need one label per Digit
+    // can use nDigit output of processHit
+    for ( int i=0; i<ndigits; ++i ) {
```

Comment:
ok, I'm getting confused here. 
First, I would change the name of the data member mMCTruthContainer to make it clearer what it is a container for, e.g. something like mTrackLabels ?
Next, the label construction should happen outside of the loop.

Reply:
Ok, I will have a look later.

Reply:
Ok, for the generation, sorry. 
I renamed it also now. 
Is there something else, which is strange to you?

Reply:
nope. looks good to me.

Reply:
Ok, shall I do a PR with this already or do we wait for other developments?
About pile-up, I wrote a novel on JIRA, so might have a look and let me know what are the time scales.

Reply:
You can do a PR. Not 100% sure I’ll be able to review it next week, we’ll see.
For your novel, I’ll read it after my holidays ;-)